resolvers/ambulance.js

```diff
@@ -68,20 +68,27 @@ const ambulanceResolvers = {
                 args.id
             );
           }
-        });
-
-      await db.eventAmbulances.destroy({
-        where: {
-          ambulanceId: args.id,
-        },
-      });
-
-      return db.ambulance.destroy({
-        where: {
-          id: args.id,
-        },
-      });
-    },
+        })
+        .then(() =>
+          db.eventAmbulances.destroy({
+            where: {
+              ambulanceId: args.id,
+            },
+          })
+        )
+        .then(() =>
+          db.ambulance.destroy({
+            where: {
+              id: args.id,
+            },
+          })
+        )
+        .then((isDeleted) => {
+          if (isDeleted === 1) {
+            return args.id;
+          }
+          throw new Error('Deletion failed for ambulance ID: ' + args.id);
```

Comment:
We have no way of knowing why it failed at this point, so that's why the error message is pretty general.